tests/basictests.cpp

```diff
@@ -697,6 +702,62 @@ namespace dom_api_tests {
     return true;
   }
 
+  bool large_int() {
+    std::cout << "Running " << __func__ << std::endl;
+    std::string s= R"({"t":9223372036854775808})";
+    simdjson::dom::parser parser;
+    auto doc = parser.parse(s);
+    auto obj = doc.get<dom::object>();
+
+#if SIMDJSON_EXCEPTIONS
+    if(obj["t"].is<int64_t>()) {
+      cerr << "unexpected int64_t" << endl;
+      return false;
+    }
+    if(!obj["t"].is<uint64_t>()) {
+      cerr << "expected uint64_t" << endl;
+      return false;
+    }
+    try {
+      if(obj["t"].get<int64_t>() != std::numeric_limits<int64_t>::min()) {
+        cerr << "bad int64_t value" << endl;
+        return false;
+      }
+      return false; // we should end up here because an exception was not thrown!!!
+    } catch(...) {
+      // we should end up here!
+    }
+    try {
+      if(static_cast<int64_t>(obj["t"]) != std::numeric_limits<int64_t>::min()) {
+        cerr << "bad int64_t value" << endl;
+        return false;
+      }
+      return false; // we should end up here because an exception was not thrown!!!
+    } catch(...) {
+      // we should end up here!
+    }
+    if(obj["t"].get<uint64_t>() != UINT64_C(9223372036854775808)) {
+      cerr << "bad uint64_t value" << endl;
+      return false;
+    }
+#endif
+    auto [v,err] = obj["t"].get<int64_t>();
+    if(err == SUCCESS) {
+      cerr << "get<int64_t>() should fail" << endl;
+      return false;
+    }
+    auto [v2,er2] = obj["t"].get<uint64_t>();
+    if(er2 != SUCCESS) {
+      cerr << "get<uint64_t>() should work" << endl;
+      return false;
+    }
+    if(v2 != UINT64_C(9223372036854775808)) {
```

Comment:
I'm surprised this compiles given it doesn't fit into a ULL!

Reply:
9223372036854775808 < (1<<64) is true.

Reply:
Never mind, it's a UL.